gwinferno/cosmology.py

```diff
@@ -45,10 +53,25 @@ def __init__(self, Ho, omega_matter, omega_radiation, omega_lambda, distance_uni
         self.OmegaLambda = omega_lambda
         self.OmegaKappa = 1.0 - (self.OmegaMatter + self.OmegaRadiation + self.OmegaLambda)
         assert self.OmegaKappa == 0, "we only implement flat cosmologies! OmegaKappa must be 0"
-        self.z = jnp.array([0.0])
-        self.Dc = jnp.array([0.0])
-        self.Vc = jnp.array([0.0])
-        self.extend(max_z=2.3, dz=DEFAULT_DZ)
+
+        if not os.path.exists(gwp + f"_{distance_unit}.h5"):
```

Comment:
I think this caching scheme as it's written is _really_ prone to bugs.  From what I can see, the only thing that would trigger a rewriting of the cache file is a change `distance_unit`, while in practice there are several things that the user could change where they would expect a new cache to be written.

The whole while loop method of extending the redshift / comoving volume / ... seems like overkill... Don't we almost exclusively use a maximum redshift?  It seems like we could do this without a need for loops.

Reply:
Removed this and was able to significantly speed up integral calculations by jaxify-ing them in [92431e7](https://github.com/FarrOutLab/GWInferno/pull/70/commits/92431e70ca8147b65520aedff9b46188bed72518).